parser/json/parser.go

```diff
@@ -229,10 +241,12 @@ func (p *parser) parseObject() (*value.Value, error) {
 
 		// Parse out the value.
 		p.skipWS()
+		p.depth++
 		v, err := p.parseValue()
```

Comment:
I was expecting to see another `p.depth++` when it enters a nested obj inside of `p.parseValue()`.

Reply:
Hm not sure if I follow, so the maximum depth only applies to JSON objects, in other words the only time `depth` may be incremented is in `parseObject`, which is where `depth` is currently incremented. Specifically, `parseObject` will call `parseValue` to parse out the values for each key, and for values that are objects, `parseValue` will in turn call `parseObject` which increments `depth`.

There's also a unit test `TestParseValueMaxDepth` to test various cases.

Reply:
Woops, the way github collapsed the diff it looked like this was inside of `Parse()` XD.